### ISSUE TITLE
The order of dropout and *resweight

### ISSUE THREAD

**Issue by OneDirection9** [majumderb/rezero#9]
Hi, nice work.

I notice that in `encoder` layer, you multiply by resweight and then do dropout. But in `decoder` layer, you do dropout and then multiply by resweight. Does the order of dropout and *resweight matter?

Thanks!

**Comment by bingone** [comment]
@OneDirection9  hi，Do you have an answer？

**Comment by OneDirection9** [comment]
@bingone Hi, I think the order is not matter.

1. `Dropout(W * res_weight)`
2. `res_weight * Dropout(W)`

According to [PyTorch doc](https://pytorch.org/docs/stable/nn.html?highlight=dropout#torch.nn.Dropout), `Dropout` is zeroes some of elements randomly. When we dropout the same elements, the `1` is equal to `2`, since `0 * x = 0`.

Thanks.

**Comment by bingone** [comment]
Thanks for your reply, I understand.
🎉🎉🎉
